src/control/lib/netdetect/netdetect.go

```diff
@@ -582,7 +581,10 @@ func GetNUMASocketIDForPid(ctx context.Context, pid int32) (int, error) {
 
 	cpuset := C.hwloc_bitmap_alloc()
 	defer C.hwloc_bitmap_free(cpuset)
+
+	mutex.Lock()
```

Comment:
Hmm. It seems reasonable to me to assume that pretty much all hwloc operations are not thread-safe... The fact that the other unguarded calls haven't caused problems (yet) may be due to luck rather than because they're safe.

I see two ways to address this, both of which are going to require some code reorganization:
  1) Expand the use of the mutex in this struct to cover more/all of the calls to hwloc, being careful not to introduce deadlocks
  2) Force all interactions with hwloc to happen in a single goroutine, using channels for communication between it and the non-hwloc goroutines

The second option may not be ideal, both in terms of the amount of work necessary to achieve it and also in terms of code clarity, but I figured I'd bring it up as a possibility.

Reply:
> 
> 
> It might make sense to add a thin Go wrapper for hwloc to manage those accesses with the mutex, so netdetect doesn't need to worry about that anymore. I would probably create a new hwLoc type to contain that interface.

Yeah that might make sense too. I think the netdetect library started off as a relatively thin wrapper around hwloc, but it's not so thin anymore. In any case, some rethinking and refactoring is probably warranted here. I'm concerned that this reactive approach is going to drag out as we find more concurrency bugs.

Reply:
I took a crack at implementing Kris's suggestion.  We discussed offline that I'll file another ticket to investigate the possibility of other hwloc concurrency issues, rather than try to address all of that in this PR.